base/src/main/java/butter/droid/base/manager/provider/ProviderManager.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of Butter.
+ *
+ * Butter is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Butter is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butter. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package butter.droid.base.manager.provider;
+
+import android.support.annotation.IntDef;
+import android.support.annotation.MainThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.ArrayList;
+import java.util.List;
+
+import butter.droid.base.providers.media.MediaProvider;
+import butter.droid.base.providers.subs.SubsProvider;
+
+public class ProviderManager {
+
+    // region IntDef
+
+    @IntDef({PROVIDER_TYPE_MOVIE, PROVIDER_TYPE_SHOW})
+    @Retention(RetentionPolicy.SOURCE)
+    public @interface ProviderType {
+    }
+
+    public static final int PROVIDER_TYPE_MOVIE = 0;
+    public static final int PROVIDER_TYPE_SHOW = 1;
+
+    // endregion IntDef
+
+    @Nullable private final MediaProvider movieProvider;
+    @Nullable private final MediaProvider showProvider;
+    @Nullable private final SubsProvider subsProvider;
+
+    @ProviderType private int currentProviderType;
+
+    private final List<OnProviderChangeListener> listeners = new ArrayList<>();
+
+    public ProviderManager(@Nullable MediaProvider movieProvider, @Nullable MediaProvider showProvider,
+            @Nullable SubsProvider subsProvider) {
+        this.movieProvider = movieProvider;
+        this.showProvider = showProvider;
+        this.subsProvider = subsProvider;
```

Comment:
This way only one `SubsProvider` will be able to be used, but it should be able to differ based on the actual `MediaProvider`.